R/utils.R

```diff
@@ -43,7 +43,7 @@ convert_cmdstan_to_rstan <- function(functions) {
   )
   #   case 2: array[n] real x -> real x[n]
   functions <- gsub(
-    "array\\[(.*)\\] (.*) ([a-z_]+)", "\\2 \\3[\\1]", functions
+    "array\\[([^]]*)\\]\\s+(.*)\\s+([a-z_]+)", "\\2 \\3[\\1", functions
```

Comment:
missing a closing `]` before the last quotation mark

Also I think you need to replace `(.*)` with `([a-z_]*)` otherswise you get the same greedy matching issues as before.

Reply:
That won't account properly for vector variables will it?

Reply:
I must be missing something. Can you give an example where it wouldn't?

Reply:
I've added another line of regex to crudely deal with this. I think perhaps should fix it as bugs occur now? Also added a few harder tests for things not currently in functions but that could be. See https://github.com/epiforecasts/epinowcast/pull/93/commits/5953dd18d7389499a84a218defe483212d4e04ba